storage/badger/guarantees.go

```diff
@@ -35,7 +35,7 @@ func NewGuarantees(collector module.CacheMetrics, db *badger.DB) *Guarantees {
 	g := &Guarantees{
 		db: db,
 		cache: newCache(collector, metrics.ResourceGuarantee,
-			withLimit(flow.DefaultTransactionExpiry+100),
+			withLimit(cacheSize),
```

Comment:
it seems we had some heuristic on how to compute the cache size for Guarantees, is there anywhere else in the code we keep this beside the default value ?

Reply:
same question here, we might have a reason to choose flow.DefaultTransactionExpiry+100, what was that? And is the new value higher than it?

Reply:
Comment says ```// DefaultTransactionExpiry is the default expiry for transactions, measured
// in blocks. Equivalent to 10 minutes for a 1-second block time.
const DefaultTransactionExpiry = 10 * 60```
which is a bit wrong (as we do blocks every 2 seconds). But I don't see correlation between guarantee cache size and transaction expiry though.
However, previously this would `60*10 + 100 = 700` so default `1000` is indeed higher

Reply:
i don't remember any specific reason for using the transaction expiry time for this cache size, do you @jordanschalm ?

Either way, i think it's fine for it to be a bit higher.

Reply:
We used `DefaultTransactionExpiry` for the cache size of block headers and contents, because the retrieval pattern for consensus nodes was to walk back `DefaultTransactionExpiry` many blocks to find duplicate transactions. So `DefaultTransactionExpiry+100` was that value + a buffer. For other caches I think this was just copied -- definitely good to select different values for each entity type.